### PR TITLE
[feat]: [CDS-70000]: Adding the logic for override v2 settings update

### PR DESCRIPTION
**Pull request by lovish1999**
## Describe your changes
Adding the logic for override v2 settings update API & corresponding UTs. We follow the similar implementation we had for migration api i.e. the user can either set/update the settings at a single scope or for entities under a scope. We use the existing NgSettingsClient method to update the settings database. 

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/48583)

### CODE REVIEW

File: 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/helpers/serviceoverridesv2/services/ServiceOverrideV2SettingsUpdateServiceImpl.java [hunk 1]
```diff
@@ -7,16 +7,275 @@
...
+      String accountId, boolean isRevert) {
+    boolean isSettingsUpdateSuccessful = true;
+    List<SettingResponseDTO> responseDTOCall = null;
+    try {
+      responseDTOCall = getResponseDTOCall(accountId, null, null, isRevert);
+    } catch (Exception e) {
+      isSettingsUpdateSuccessful = false;
+      log.error(
+          String.format(DEBUG_LINE + "Override settings update failed for project with accountId: [%s]", accountId), e);
+    }
```
Comment (tchaurasiya) [inline 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/helpers/serviceoverridesv2/services/ServiceOverrideV2SettingsUpdateServiceImpl.java:273]: "failed for project with accountId" typo? its account setting right?

**Review by tchaurasiya — CHANGES_REQUESTED** [review]

**Review by lovish1999 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/48583)* status: 8 file(s) to review. 1 change requests. 9. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`120-ng-manager/src/main/java/io/harness/ng/core/serviceoverrides/resources/ServiceOverridesResource.java` line 364 at r2](https://reviewable.io/reviews/harness/harness-core/48583#gh-1211590852:-NWq0ESfDiihLL_HPP_h:b-qixzjg) ([raw file](https://github.com/harness/harness-core/blob/094b20ccd6ecaac6902eed445a27ab072b9e81a4/120-ng-manager/src/main/java/io/harness/ng/core/serviceoverrides/resources/ServiceOverridesResource.java#L364)):*
<details><summary><i>Previously, tchaurasiya (Tathagat Chaurasiya) wrote…</i></summary><blockquote>

seems like we are passing opposite boolean value for updateChildren field
</blockquote></details>

Ok, it was in sync with the migrations apis. Making this change in all Migration & revert APIs
___
*[`120-ng-manager/src/main/java/io/harness/ng/core/serviceoverrides/resources/ServiceOverridesResource.java` line 392 at r2](https://reviewable.io/reviews/harness/harness-core/48583#gh-1211592471:-NWq05lm5RAZPdGntkwU:b-qixzjg) ([raw file](https://github.com/harness/harness-core/blob/094b20ccd6ecaac6902eed445a27ab072b9e81a4/120-ng-manager/src/main/java/io/harness/ng/core/serviceoverrides/resources/ServiceOverridesResource.java#L392)):*
<details><summary><i>Previously, tchaurasiya (Tathagat Chaurasiya) wrote…</i></summary><blockquote>

seems like we are passing opposite boolean value for updateChildren field 
</blockquote></details>

Ok, it was in sync with the migrations apis. Making this change in all Migration & revert APIs
___
*[`120-ng-manager/src/main/java/io/harness/ng/core/serviceoverrides/resources/ServiceOverridesResource.java` line 415 at r2](https://reviewable.io/reviews/harness/harness-core/48583#gh-1211594037:-NWpydMq6xrepdAjo9O3:b32wtd) ([raw file](https://github.com/harness/harness-core/blob/094b20ccd6ecaac6902eed445a27ab072b9e81a4/120-ng-manager/src/main/java/io/harness/ng/core/serviceoverrides/resources/ServiceOverridesResource.java#L415)):*
<details><summary><i>Previously, tchaurasiya (Tathagat Chaurasiya) wrote…</i></summary><blockquote>

lets use primitive boolean
</blockquote></details>

ok
___
*[`125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/helpers/serviceoverridesv2/services/ServiceOverrideV2SettingsUpdateServiceImpl.java` line 40 at r2](https://reviewable.io/reviews/harness/harness-core/48583#gh-1211600614:-NWpynqyEU8jGLWFm4U1:b32wtd) ([raw file](https://github.com/harness/harness-core/blob/094b20ccd6ecaac6902eed445a27ab072b9e81a4/125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/helpers/serviceoverridesv2/services/ServiceOverrideV2SettingsUpdateServiceImpl.java#L40)):*
<details><summary><i>Previously, tchaurasiya (Tathagat Chaurasiya) wrote…</i></summary><blockquote>

nit: lets mark them private
</blockquote></details>

ok
___
*[`125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/helpers/serviceoverridesv2/services/ServiceOverrideV2SettingsUpdateServiceImpl.java` line 198 at r2](https://reviewable.io/reviews/harness/harness-core/48583#gh-1211625145:-NWpznR_BMxZ8PV7Tv_I:b32wtd) ([raw file](https://github.com/harness/harness-core/blob/094b20ccd6ecaac6902eed445a27ab072b9e81a4/125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/helpers/serviceoverridesv2/services/ServiceOverrideV2SettingsUpdateServiceImpl.java#L198)):*
<details><summary><i>Previously, tchaurasiya (Tathagat Chaurasiya) wrote…</i></summary><blockquote>

can we rename this, to something like  updateOverrideSetting (current name is not telling what this method is doing)
</blockquote></details>

ok
___
*[`125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/helpers/serviceoverridesv2/services/ServiceOverrideV2SettingsUpdateServiceImpl.java` line 214 at r2](https://reviewable.io/reviews/harness/harness-core/48583#gh-1211623409:-NWpzorZBgN6IyEpWkVX:b2nign6) ([raw file](https://github.com/harness/harness-core/blob/094b20ccd6ecaac6902eed445a27ab072b9e81a4/125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/helpers/serviceoverridesv2/services/ServiceOverrideV2SettingsUpdateServiceImpl.java#L214)):*
<details><summary><i>Previously, tchaurasiya (Tathagat Chaurasiya) wrote…</i></summary><blockquote>

Please make sure to test this with account/org level settings.
</blockquote></details>

yes
___
*[`125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/helpers/serviceoverridesv2/services/ServiceOverrideV2SettingsUpdateServiceImpl.java` line 219 at r2](https://reviewable.io/reviews/harness/harness-core/48583#gh-1211614870:-NWq-3xHASC3OqVgyDwo:b-bs9fj0) ([raw file](https://github.com/harness/harness-core/blob/094b20ccd6ecaac6902eed445a27ab072b9e81a4/125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/helpers/serviceoverridesv2/services/ServiceOverrideV2SettingsUpdateServiceImpl.java#L219)):*
<details><summary><i>Previously, tchaurasiya (Tathagat Chaurasiya) wrote…</i></summary><blockquote>

nit : generally we should initialize list with new ArrayList()<> instead of null
</blockquote></details>

ok, yes. 
___
*[`125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/helpers/serviceoverridesv2/services/ServiceOverrideV2SettingsUpdateServiceImpl.java` line 250 at r2](https://reviewable.io/reviews/harness/harness-core/48583#gh-1211626356:-NWq04CeEizyszKUUBTB:b-xbh3ld) ([raw file](https://github.com/harness/harness-core/blob/094b20ccd6ecaac6902eed445a27ab072b9e81a4/125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/helpers/serviceoverridesv2/services/ServiceOverrideV2SettingsUpdateServiceImpl.java#L250)):*
<details><summary><i>Previously, tchaurasiya (Tathagat Chaurasiya) wrote…</i></summary><blockquote>

"settings update failed for project with orgId" typo?
</blockquote></details>

yes, correcting it. 
___
*[`125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/helpers/serviceoverridesv2/services/ServiceOverrideV2SettingsUpdateServiceImpl.java` line 273 at r2](https://reviewable.io/reviews/harness/harness-core/48583#gh-1211627750:-NWq02g2CvQXBVZPyoN5:b-xbh3ld) ([raw file](https://github.com/harness/harness-core/blob/094b20ccd6ecaac6902eed445a27ab072b9e81a4/125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/service/steps/helpers/serviceoverridesv2/services/ServiceOverrideV2SettingsUpdateServiceImpl.java#L273)):*
<details><summary><i>Previously, tchaurasiya (Tathagat Chaurasiya) wrote…</i></summary><blockquote>

"failed for project with accountId" typo? its account setting right?
</blockquote></details>

yes, correcting it.

**Review by tchaurasiya — APPROVED** [review]
> Reviewed 1 of 3 files at r2.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/48583)* status: 8 file(s) to review. 1 of 2 approvals obtained.

**Comment by lovish1999** [comment]
trigger uts

**Comment by lovish1999** [comment]
trigger uts

**Comment by lovish1999** [comment]
trigger uts

**Comment by lovish1999** [comment]
trigger uts

**Comment by lovish1999** [comment]
tigger uts

**Comment by lovish1999** [comment]
trigger uts

**Comment by lovish1999** [comment]
trigger uts

**Review by tiwaritejas — APPROVED** [review]

**Comment by lovish1999** [comment]
trigger ut3

**Comment by lovish1999** [comment]
trigger review

**Review by yogesh-chauhan — APPROVED** [review]
:lgtm:

> Reviewed 5 of 8 files at r1, 1 of 3 files at r2, 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/48583#-:-NX-SdbZ1r78cUOvqWNP:bnfp4nl)* status: 7 file(s) to review.

**Pull request merged**